plugins/zfs/zfsonlinux_stats_

```diff
@@ -0,0 +1,330 @@
+#!/bin/sh
+# ZFS statistics for ZFSonLinux
+# Author: Adam Michel (elfurbe@furbism.com)
+#
+# Description:
+#  This is a modification of the zfs_stats
+#  plugin by David Bjornsson (which was a
+#  rewrite of zfs-stats-for-freebsd scripts
+#  by patpro) modified to work with ZFSonLinux.
+#
+# Tested on Ubuntu-14.04
+#
+# Usage: zfs_stats_FUNCTION
+#
+# Available functions:
+#	efficiency - ARC efficiency
+#	cachehitlist - Cache hit by cache list
+#	cachehitdtype - Cache hit by data type
+#	dmuprefetch - DMU prefetch
+#	utilization - ARC size breakdown
+#	l2utilization - L2ARC size breakdown
+#	l2efficiency - L2ARC efficiency
+#
+#%# family=auto
+
+FUNCTION=$(basename $0 | cut -d_ -f3)
+MEMMAX=`cat /proc/meminfo | grep MemTotal | awk '{print $2}'`
+BC='/usr/bin/bc -q'
+ARCSTATS="/proc/spl/kstat/zfs/arcstats"
+ZFETCHSTATS="/proc/spl/kstat/zfs/zfetchstats"
+
+ARC_HITS=`cat $ARCSTATS | grep ^hits | awk '{print $3}'`
```

Comment:
There are some inefficiencies here, which should be addressed.

First, the pattern "cat $ARCSTATS | grep ^bar | awk {print $2}" can be done as a single "awk '$1 == "bar" {print $3}' $ARCSTATS" per line.

The plugin would also improve if you find more ways to avoid forking. Perhaps the arcstats should be parsed in its entirety with shell (cat $ARCSTATS | while read name something value; do ... ; done), alternatively awk, or perl, without calling out to external utilities?

Thanks for the pull request.  A good ZFS plugin would be welcome.